File: LLM-Translations/GPT-4/codenet/Go/Java/s864554008.java

import java.util.Scanner;

public class s864554008 {
    public static void main(String[] args) {
        Data d = newData();
        if (d.exist()) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }
    }

    private static Data newData() {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        String[] nImg = load(scanner, n);
        String[] mImg = load(scanner, m);
        return new Data(n, m, nImg, mImg);
    }

    private static String[] load(Scanner scanner, int cnt) {
        String[] ret = new String[cnt];
        for (int i = 0; i < cnt; i++) {
            ret[i] = scanner.next();
        }
        return ret;
    }
}

class Data {
    int n, m;
    String[] nImg, mImg;

    Data(int n, int m, String[] nImg, String[] mImg) {
        this.n = n;
        this.m = m;
        this.nImg = nImg;
        this.mImg = mImg;
    }

    boolean exist() {
        for (int i = 0; i <= n - m; i++) {
            for (int j = 0; j <= n - m; j++) {
                if (match(i, j)) {
                    return true;
                }
            }
        }
        return false;
    }

    boolean match(int x, int y) {
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < m; j++) {
                if (nImg[x + i].charAt(y + j) != mImg[i].charAt(j)) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: LLM-Translations/GPT-4/codenet/Go/C/s864554008.c
#include <stdio.h>
#include <stdbool.h>

typedef struct {
    int n, m;
    char **nImg, **mImg;
} Data;

Data newData();
bool exist(Data d);
bool match(Data d, int x, int y);
char** load(int cnt);

int main() {
    Data d = newData();
    if (exist(d)) {
        printf("Yes\n");
    } else {
        printf("No\n");
    }
    return 0;
}

Data newData() {
    int n, m;
    scanf("%d %d", &n, &m);
    char **nImg = load(n);
    char **mImg = load(m);

    Data d = {n: n, m: m, nImg: nImg, mImg: mImg};
    return d;
}

char** load(int cnt) {
    char **ret = (char**)malloc(cnt * sizeof(char*));
    for (int i = 0; i < cnt; i++) {
        ret[i] = (char*)malloc((cnt + 1) * sizeof(char));
        scanf("%s", ret[i]);
    }
    return ret;
}

bool exist(Data d) {
    for (int i = 0; i <= d.n - d.m; i++) {
        for (int j = 0; j <= d.n - d.m; j++) {
            if (match(d, i, j)) {
                return true;
            }
        }
    }
    return false;
}

bool match(Data d, int x, int y) {
    for (int i = 0; i < d.m; i++) {
        for (int j = 0; j < d.m; j++) {
            if (d.nImg[x + i][y + j] != d.mImg[i][j]) {
                return false;
            }
        }
    }
    return true;
}
// 
